packages/driver/src/util/config.ts

```diff
@@ -25,7 +25,7 @@ const omitConfigReadOnlyDifferences = (objectLikeConfig: Cypress.ObjectLike) =>
       return
     }
 
-    if (overrideLevels === 'never') {
+    if (overrideLevels === 'never' && configKey !== 'protocolEnabled') {
```

Comment:
Technically this function is redundant for setting state for cy.orgin since the validateConfig handler ensures the override levels, but this changes allows us to tests this logic in cy.origin.